google/cloud/aiplatform/metadata/resource.py

```diff
@@ -187,35 +189,69 @@ def get_or_create(
             )
         return resource
 
+    def sync_resource(self):
+        """Syncs local resource with the resource in metadata store."""
+        self._gca_resource = getattr(self.api_client, self._getter_method)(
+            name=self.resource_name, retry=base._DEFAULT_RETRY
+        )
+
+    @staticmethod
+    def _nested_update_metadata(
+        gca_resource: Union[
+            gca_context.Context, gca_execution.Execution, gca_artifact.Artifact
+        ],
+        metadata: Optional[Dict[str, Any]] = None,
+    ):
+        """Helper method to update gca_resource in place.
+
+        Performs a one-level deep nested update on the metadata field.
+
+        Args:
+            gca_resource (Union[gca_context.Context, gca_execution.Execution, gca_artifact.Artifact]):
+                Required. Metadata Protobuf resource. This proto's metadata will be
+                updated in place.
+            metadata (Dict[str, Any]):
+                Optional. Metadata dictionary to merge into gca_resource.metadata.
+        """
+
+        if metadata:
+            if gca_resource.metadata:
+                for key, value in metadata.items():
+                    # Note: This only support nested dictionaries one level deep
+                    if isinstance(value, collections.Mapping):
+                        gca_resource.metadata[key].update(value)
```

Comment:
What is the proposed way to remove keys from a nested dictionary?

P.S. This also seems to fail when the `key` does not already exist in the metadata.